src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -321,14 +352,55 @@ private void checkClientServerCompatibility(
   }
 
   @Override
-  public void afterCommand() {
+  public void afterCommand() throws AbruptExitException {
+    IOException failure = null;
+    try {
+      deletedDownloadedInputs();
+    } catch (IOException e) {
+      failure = e;
+    }
+
+    try {
+      closeRpcLogFile();
+    } catch (IOException e) {
+      failure = e;
+    }
+
     buildEventArtifactUploaderFactoryDelegate.reset();
     actionContextProvider = null;
+    actionInputFetcher = null;
+
+    if (failure != null) {
+      throw new AbruptExitException(ExitCode.LOCAL_ENVIRONMENTAL_ERROR, failure);
+    }
+  }
+
+  /**
+   * Delete any input files that have been fetched from the remote cache during the build. This
+   * is so that Bazel's view of the output base is identical with the output base after a build
+   * i.e. files that Bazel thinks exist only remotely actually do.
+   */
+  private void deletedDownloadedInputs() throws IOException {
+    if (actionInputFetcher == null) {
+      return;
+    }
+    IOException deletionFailure = null;
+    for (Path file : actionInputFetcher.downloadedFiles()) {
+      try {
+        file.delete();
+      } catch (IOException e) {
+        deletionFailure = e;
```

Comment:
In case 2 files cannot be deleted for some reason, we will throw only an exception for the second file. The first one will be ignored. Is that an expected behavior?

Reply:
I added log statements. I think we mostly care about being able to delete the files staged in the output base. If flushing the rpc log fails, it's not a big deal.

Reply:
I think this snippet is not doing anything with the rpc logs. 
Let's assume there are 5 downloaded files to be deleted: file1, file2,...,file5. We iterate over those files (in the order from file1 to file5) and `try` to delete them. Now, we got exceptions for the files file1 and file4, others were removed successfully. Based on the current implementation, we will propagate further only the file4 related exception, exception related to the file1 will be ignored.

Reply:
Ooops I looked at the wrong code. Yes, it's intentional to only report one error to the user. First of all, it's highly unlikely that there will ever be an error deleting these files since the output base is the one directory that Bazel controls and if we don't have write access there we'll crash way before this code every runs. Secondly, if there is a problem deleting a few files it will likely be the same problem for all files. Thirdly, there will thousands or tens of thousands of files and so we can't possibly report each individual error. Nevertheless, I added log statements to the code so at least each message will show up in the log. Makes sense?

Reply:
I see. Ok, makes sense now, thanks!